### DISCUSSION TITLE
New Deduplicator feature, replace with hard link

### DISCUSSION THREAD

**Discussion by Raffael7777** [d4rken-org/sdmaid-se#1184] [Ideas]
**Is your feature request related to a problem? Please describe.**
Often duplicate files are a necessary evil, as different applications might both expect the identical files to be in a specific spots, so deletion would break one of the apps functionality. However, that doesn't mean we can't do anything about it.

**Describe the solution you'd like**
Provided the phone's file system supports hard links, I would like SD Maid to offer the option within the Deduplicator to replace any number of selected duplicates (up to all but one) with hard links to an identical source file of the user's selection.

**Describe alternatives you've considered**
This feature is available as several commands within Linux, and a handful of Windows applications, so I could plug my phone into my PC and point said app at the device's storage, however this is not convenient, and is slow due to my phone's non-USB-3speed C port, and I believe more users should learn about this space saving option.

**Describe why this would be in the interest of all users**
In certain situations where you **must** have duplicate files, hard links can save considerable space with no effect on how the file is accessed or used by various apps. Not as much as deletion, but close to.

**Additional context**
Not all file systems support hard links, and such an option might need to be hidden behind an "advanced" menu, or users should be prefaced to convey the bare minimum of necessary information about how hard links function. While the file is not truly deleted until the last hard link is deleted (the original file is technically a hard link itself), it is crucial to understand that edits to one hard link are made to all hard links that point to the same file inode, so users should not replace a duplicate file with a hard link if they or an app in the future might want to change one of the files and not the others.

PS: Let us be able to type in the minimum file size in the Deduplicator settings, the slider is difficult to use to select specific cutoffs.

**Comment by d4rken** [comment]
I think this might be a bit too specific and only relevant for a very small number of rooted power users while also being fairly complex to implement and having some potential to screw up a lot of apps.

I'm currently thinking this is out of scope, but will transfer it to discussions.
